samples/subsys/ipc/openamp_rsc_table/src/main_remote.c

```diff
@@ -11,7 +11,15 @@
 #include <stdlib.h>
 #include <string.h>
 
+#if !defined(CONFIG_IPM) && !defined(CONFIG_MBOX)
+#error Either CONFIG_IPM or CONFIG_MBOX must be enabled
+#endif
+
+#ifdef CONFIG_IPM
```

Comment:
TBH I don't like this 
```
#ifdef
...
#else
..
#endif
```
There is a [driver to convert from mbox to ipm](https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/ipm/ipm_mbox.c), I'm using it for 8ULP (https://github.com/zephyrproject-rtos/zephyr/pull/83049/commits).

@arnopo @tnmysh what do you think?
IMO we should use the `mbox-to-ipm` driver since the IPM is not completely deprecated, and is still used in other samples.